firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/core/CrashlyticsController.java

```diff
@@ -858,6 +858,9 @@ private void doCloseSessions(int maxCustomExceptionEvents, boolean includeCurren
 
     if (includeCurrent) {
       reportingCoordinator.onEndSession();
+    } else if (nativeComponent.hasCrashDataForSession(mostRecentSessionIdToClose)) {
```

Comment:
can you add some justification for why this is an else if? why do we only do native stuff if includeCurrent = false? Why do we not do it if includeCurrent = false?

Reply:
Yeah, this is one of those "the code needs more cleaning" issues. `includeCurrent` implies this is being run at Java crash time. We only want to run this code for the startup-time initialize.

Reply:
Added a comment.